### ISSUE TITLE
Running setup.py install for pillow ... error

### ISSUE THREAD

**Issue by hot36952** [Flexget/Flexget#2646]
What can I do？

./python3` -m pip install flexget

```
> 
Installing collected packages: pytz, tzlocal, webencodings, werkzeug, terminaltables, html5lib, babelfish, python-dateutil, rebulk, guessit, progressbar, pyyaml, zxcvbn-python, zc.lockfile, aniso8601, markupsafe, jinja2, itsdangerous, click, flask, flask-restful, colorclass, certifi, plumbum, rpyc, apscheduler, pyrss2gen, python-xlib, pillow, pystray, tempora, jsonschema, flask-restplus, chardet, idna, pyparsing, more-itertools, jaraco.functools, urllib3, loguru, portend, requests, pynzb, flask-compress, soupsieve, beautifulsoup4, flask-login, sqlalchemy, feedparser, flask-cors, cheroot, cherrypy, flexget
    Running setup.py install for tzlocal ... done
    Running setup.py install for terminaltables ... done
    Running setup.py install for babelfish ... done
    Running setup.py install for rebulk ... done
    Running setup.py install for guessit ... done
    Running setup.py install for progressbar ... done
    Running setup.py install for pyyaml ... done
    Running setup.py install for zxcvbn-python ... done
    Running setup.py install for aniso8601 ... done
    Running setup.py install for markupsafe ... done
    Running setup.py install for itsdangerous ... done
    Running setup.py install for colorclass ... done
    Running setup.py install for rpyc ... done
    Running setup.py install for pyrss2gen ... done
  Attempting uninstall: pillow
    Found existing installation: Pillow 6.2.1
    Uninstalling Pillow-6.2.1:
      Successfully uninstalled Pillow-6.2.1
    Running setup.py install for pillow ... error
    ERROR: Command errored out with exit status 1:
     command: /opt/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-0yg4me7x/pillow/setup.py'"'"'; __file__='"'"'/tmp/pip-install-0yg4me7x/pillow/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-1eloatkz/install-record.txt --single-version-externally-managed --compile --install-headers /opt/include/python3.8/pillow
         cwd: /tmp/pip-install-0yg4me7x/pillow/
    Complete output (162 lines):
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-aarch64-3.8
    creating build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/FtexImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/PpmImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ImageFont.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/JpegImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/PsdImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ImImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/SgiImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/XVThumbImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/__main__.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/Image.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/_binary.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/PyAccess.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ImageSequence.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/DdsImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/XpmImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/McIdasImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/IcnsImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ImageTransform.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/BmpImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ImagePalette.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/IptcImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/TiffImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/GdImageFile.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/_util.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/CurImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/TiffTags.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/FliImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/WebPImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ImageGrab.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/WmfImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ImageEnhance.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ImageShow.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/PdfImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ImageOps.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/GbrImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ImageTk.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/BlpImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/PcfFontFile.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/WalImageFile.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/Jpeg2KImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ImageMorph.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/TgaImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/EpsImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/_tkinter_finder.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/IcoImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ImageWin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/GimpGradientFile.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ImageDraw.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/features.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/Hdf5StubImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/JpegPresets.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/__init__.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/DcxImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ImagePath.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ImageDraw2.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ImageFilter.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/PcdImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/PSDraw.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/BufrStubImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ImageChops.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/FpxImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ImageCms.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ExifTags.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/PdfParser.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/FontFile.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/PaletteFile.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ImageMath.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ImageQt.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/PalmImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/GifImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ImtImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/_version.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/MicImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/MspImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ContainerIO.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ImageColor.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/GimpPaletteFile.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/PixarImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ImageStat.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/GribStubImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/BdfFontFile.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/MpegImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/FitsStubImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/TarIO.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/SpiderImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ImageFile.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/SunImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/PcxImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/XbmImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/PngImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/ImageMode.py -> build/lib.linux-aarch64-3.8/PIL
    copying src/PIL/MpoImagePlugin.py -> build/lib.linux-aarch64-3.8/PIL
    running egg_info
    writing src/Pillow.egg-info/PKG-INFO
    writing dependency_links to src/Pillow.egg-info/dependency_links.txt
    writing top-level names to src/Pillow.egg-info/top_level.txt
    reading manifest file 'src/Pillow.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    warning: no files found matching '*.c'
    warning: no files found matching '*.h'
    warning: no files found matching '*.sh'
    warning: no previously-included files found matching '.appveyor.yml'
    warning: no previously-included files found matching '.coveragerc'
    warning: no previously-included files found matching '.codecov.yml'
    warning: no previously-included files found matching '.editorconfig'
    warning: no previously-included files found matching '.readthedocs.yml'
    warning: no previously-included files found matching 'azure-pipelines.yml'
    warning: no previously-included files matching '.git*' found anywhere in distribution
    warning: no previously-included files matching '*.pyc' found anywhere in distribution
    warning: no previously-included files matching '*.so' found anywhere in distribution
    no previously-included directories found matching '.azure-pipelines'
    no previously-included directories found matching '.travis'
    writing manifest file 'src/Pillow.egg-info/SOURCES.txt'
    running build_ext
    
    
    The headers or library files could not be found for zlib,
    a required dependency when compiling Pillow from source.
    
    Please see the install instructions at:
       https://pillow.readthedocs.io/en/latest/installation.html
    
    Traceback (most recent call last):
      File "/tmp/pip-install-0yg4me7x/pillow/setup.py", line 852, in <module>
        setup(
      File "/__init__.py", line 145, in setup
      File "/core.py", line 148, in setup
      File "/dist.py", line 966, in run_commands
      File "/dist.py", line 985, in run_command
      File "/install.py", line 61, in run
      File "/install.py", line 545, in run
      File "/cmd.py", line 313, in run_command
      File "/dist.py", line 985, in run_command
      File "/build.py", line 135, in run
      File "/cmd.py", line 313, in run_command
      File "/dist.py", line 985, in run_command
      File "/build_ext.py", line 343, in run
      File "/tmp/pip-install-0yg4me7x/pillow/setup.py", line 687, in build_extensions
        raise RequiredDependencyException(f)
    __main__.RequiredDependencyException: zlib
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-0yg4me7x/pillow/setup.py", line 907, in <module>
        raise RequiredDependencyException(msg)
    __main__.RequiredDependencyException:
    
    The headers or library files could not be found for zlib,
    a required dependency when compiling Pillow from source.
    
    Please see the install instructions at:
       https://pillow.readthedocs.io/en/latest/installation.html
    
    
    ----------------------------------------
  Rolling back uninstall of Pillow
  Moving to /opt/lib/python3.8/site-packages/PIL
   from /opt/lib/python3.8/site-packages/~IL
  Moving to /opt/lib/python3.8/site-packages/Pillow-6.2.1-py3.8.egg-info
   from /opt/lib/python3.8/site-packages/~illow-6.2.1-py3.8.egg-info
ERROR: Command errored out with exit status 1: /opt/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-0yg4me7x/pillow/setup.py'"'"'; __file__='"'"'/tmp/pip-install-0yg4me7x/pillow/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-1eloatkz/install-record.txt --single-version-externally-managed --compile --install-headers /opt/include/python3.8/pillow Check the logs for full command output.

```

**Comment by liiight** [comment]
which os are you on? does this still occur?

**Comment by liiight** [comment]
Fixed with #2652
